crates/ide/src/syntax_tree.rs

```diff
@@ -52,6 +52,7 @@ fn syntax_tree_for_string(token: &SyntaxToken, text_range: TextRange) -> Option<
     }
 }
 
+#[allow(clippy::redundant_locals)]
```

Comment:
This should probably be fixed instead of allowed

Reply:
That can be removed safely. Will do it in the follow-up PR

Reply:
https://github.com/rust-lang/rust-analyzer/pull/16404/commits/9527e245c9deb8c477001d840b5532fe57884d44